### PR TITLE
add hyper's BufList implementation

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

**Comment by stammw** [comment]
Nice! I'm not sure it has its place in `h3::proto` though. It will be used in `h3::frame` only.

**Comment by jxs** [comment]
yeah i was not sure about it being in `h3::proto`, moved it to `h3::frame`

File: h3/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #[deny(missing_docs)]
 pub mod quic;
 
+mod frame;
```
Comment (stammw) [inline h3/src/lib.rs:4]: Maybe we should silence all the "unused" warnings with temporary `#[allow(dead_code)]`. Then removed them once they are effectively used through public API.

File: h3/src/frame/mod.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+pub(crate) mod buf;
```
Comment (stammw) [inline h3/src/frame/mod.rs:1]: Let's keep this only accessible in the `frame` module for now.

**Pull request merged**